manifests/config.pp

```diff
@@ -59,6 +60,43 @@
 
     }
 
+    'Suse': {
+      $pamconfig_mkhomedir_check_cmd = '/usr/sbin/pam-config -q --mkhomedir | grep session:'
+      $pamconfig_check_cmd           = '/usr/sbin/pam-config -q --sss | grep session:'
+
+      if $mkhomedir {
+
+        exec { 'pam-config -a --mkhomedir':
+          path   => '/bin:/usr/bin:/sbin:/usr/sbin',
+          unless => $pamconfig_mkhomedir_check_cmd,
+        }
+
+      }
+
+      exec { 'pam-config -a --sss':
+        path   => '/bin:/usr/bin:/sbin:/usr/sbin',
+        unless => $pamconfig_check_cmd,
+      }
+
+      if $manage_nsswitch {
+
```

Comment:
Hmm, maybe this is a good reason to not manage nsswitch in the sssd module.
Can you instead put your execs / or nsswitch module settings in a profile or otherwise outside this module and in your own custom setup?

Basically, if it can't work for all supported oses, with a good set of defaults, I'm not sure it should be here...